docs/source/en/model_doc/musicgen.md

```diff
@@ -53,6 +53,10 @@ better results than greedy, thus we encourage sampling mode to be used where pos
 and can be explicitly specified by setting `do_sample=True` in the call to [`MusicgenForConditionalGeneration.generate`],
 or by overriding the model's generation config (see below).
 
+Generation is limited by the sinusoidal positional embeddings to 30 second inputs. Meaning, MusicGen cannot generate more
+than 30 seconds of audio (1506 tokens), and input audio passed by Audio-Prompted Generation contributes to this limit so,
```

Comment:
Should be 1500 according to a quick back of the envelope calculation? We have the following relationship between audio length and number of generated tokens:
```
audio_length_in_s = num_tokens / model.config.audio_encoder.frame_rate
```

Where we get the frame rate as:
https://github.com/huggingface/transformers/blob/f11518a542ca2c8d276f323ef6563afa3f1b03a7/src/transformers/models/encodec/configuration_encodec.py#L188-L190

Plugging in the numbers from the [config](https://huggingface.co/facebook/encodec_32khz/blob/4966e19f4931abbc51c4bfb92ae06bdcb53b392e/config.json#L30-L34):

```python
hop_length = 8 * 5 * 4 * 4
frame_rate = 32000 / hop_length
```
So we get a `frame_rate` of 50. For an audio length of 30s, this gives:
```python
# num_tokens = audio_length_in_s * frame_rate
num_tokens = 30 * 50
```
So 1500 tokens.

```suggestion
than 30 seconds of audio (1500 tokens), and input audio passed by Audio-Prompted Generation contributes to this limit so,
```

Reply:
This `audio length <-> token` conversion is covered in the HF MusicGen notebook: https://colab.research.google.com/github/sanchit-gandhi/notebooks/blob/main/MusicGen.ipynb#scrollTo=d75ad107-e19b-47f3-9cf1-5102ab4ae74a

Reply:
I'm not sure.

When I have the model generate 1500 tokens, I don't get a full 30 seconds of audio. After generating the audio for 1500 tokens, I get 958080 samples. For a sample rate of 32000 samples per second, that's about 29.94 seconds of audio (`958080 / 32000 = 29.94`).

I generated the samples for 1506 tokens again, expecting the number of samples to be an even 960000 like before, but this time the output samples were 961920. After playing around with it for a bit, generating the samples for 1503 tokens got me the even 960000 I was looking for, which if we divide by the sample rate of 32000 again, we get 30 seconds of samples, on the dot.

I guess I made a mistake and it's really 1503 tokens to generate 30 seconds of audio samples or it's fluctuating? I'll try playing around with it for a bit.

This is the code I used:
```python
from transformers import AutoProcessor, MusicgenForConditionalGeneration
import torch, scipy
processor = AutoProcessor.from_pretrained("facebook/musicgen-large")
with torch.no_grad():
    model = MusicgenForConditionalGeneration.from_pretrained("facebook/musicgen-large", torch_dtype=torch.float16, device_map="cuda")
inputs = processor(text=["80s pop track with bassy drums and synth"], padding=True, return_tensors="pt").to("cuda")
audio_values = model.generate(**inputs, do_sample=True, guidance_scale=3, max_new_tokens=1503).to("cpu")
```

Reply:
Feel free to update it to 1503 :)

Reply:
@sanchit-gandhi done!